File: go/leap/leap.go
// Package leap is a package for implementing the exercism.io leap exercise in Go
package leap

// IsLeapYear takes in a year and returns true if the year is a leap year and false if it is not
func IsLeapYear(year int) bool {
	if year%4 == 0 {
		if year%100 != 0 {
			return true
		} else {
			if year%400 == 0 {
				return true
			}
		}
	}
	return false
}

File: csharp/leap/Leap.cs
using System;

public static class Leap
{
    public static bool IsLeapYear(int year)
    {
        var yearDivisibleBy4 = year % 4 == 0;
        var yearDivisibleBy100 = year % 100 == 0;
        var yearDivisibleBy400 = year % 400 == 0;

        return yearDivisibleBy4 && !yearDivisibleBy100 || yearDivisibleBy400;
    }
}